raiden/network/transport/matrix/transport.py

```diff
@@ -541,7 +561,13 @@ def _send_global():
             )
             room.send_text(text)
 
-        self._spawn(_send_global)
+        greenlet = self._spawn(_send_global)
```

Comment:
One small problem here: `gevent.spawn()` immediately starts the new greenlet which means the name only gets set once control returns to the spawning context.

I'd suggest to add `name` as an arg to the `_spawn` helper and change it so it doesn't use spawn but rather instantiate, set-name, start.

Reply:
Good point! `gevent.spawn` instantiates the greelet so among kwargs we can pass `name` there according to this:
http://www.gevent.org/api/gevent.greenlet.html#gevent.Greenlet.name

Reply:
No unfortunately you can't pass it to `.spawn()`. But (at least in this pr) we're only using spawn through a helper already, so we can modify that.

Reply:
> I'd suggest to add `name` as an arg to the `_spawn` helper and change it so it doesn't use spawn but rather instantiate, set-name, start.

That was my first idea, but @andrevmatos preferred to set the name on the returned object.

Reply:
I did suggest setting the name on the returned object to preserve `_spawn` interface compatibility with `gevent.spawn` (i.e. `func, *args, **kwargs`). Even if `gevent.spawn` starts the greenlet right away, AFAIK there's no immediate context switch, so setting the name right after spawning should still guarantee the name is set on first scheduling, that's why I suggested doing it there. Anyway, breaking the interface for this internal helper wouldn't be a big deal, just thought that would be simpler